src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -162,6 +165,36 @@ class spell_rog_blade_flurry : public AuraScript
     Unit* _procTarget = nullptr;
 };
 
+// 31230 - Cheat Death
+class spell_rog_cheat_death : public AuraScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ SPELL_ROGUE_CHEAT_DEATH_DUMMY, SPELL_ROGUE_CHEATED_DEATH, SPELL_ROGUE_CHEATING_DEATH })
+            && ValidateSpellEffect({ { spellInfo->Id, EFFECT_1 } });
+    }
+
+    void HandleAbsorb(AuraEffect* /*aurEff*/, DamageInfo& dmgInfo, uint32& absorbAmount)
+    {
+        PreventDefaultAction();
+
+        if (!GetTarget()->HasAura(SPELL_ROGUE_CHEATED_DEATH))
+        {
+            int32 healAmount = int32(GetTarget()->CountPctFromMaxHealth(GetEffectInfo(EFFECT_1).CalcValue(GetTarget())));
+            GetTarget()->CastSpell(GetTarget(), SPELL_ROGUE_CHEAT_DEATH_DUMMY, true);
+            GetTarget()->CastSpell(GetTarget(), SPELL_ROGUE_CHEATED_DEATH, true);
+            GetTarget()->CastSpell(GetTarget(), SPELL_ROGUE_CHEATING_DEATH, true);
```

Comment:
You could do that, if you like

```suggestion
            for (uint32 const& spellId : { SPELL_ROGUE_CHEAT_DEATH_DUMMY, SPELL_ROGUE_CHEATED_DEATH, SPELL_ROGUE_CHEATING_DEATH })
                GetTarget()->CastSpell(GetTarget(), spellId, true);
```